webui/react/src/pages/F_ExpList/glide-table/useGlasbey.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useMemo, useState } from 'react';
+
+import { glasbeyColor } from 'shared/utils/color';
+
+export type MapOfIdsToColors = Record<string, string>;
+
+export const useGlasbey = (ids: string[]): MapOfIdsToColors => {
+  const itemIds = ids.map(String);
+  const [indexMap, setIndexMap] = useState(() =>
+    itemIds.map((itemId, idx) => ({ [itemId]: idx })).reduce((a, b) => ({ ...a, ...b }), {}),
+  );
+
+  useEffect(() => {
+    const idsInMap = Object.keys(indexMap);
+    const idsToAdd = itemIds.filter((id) => !idsInMap.includes(id));
+    const idsToRemove = idsInMap.filter((id) => !itemIds.includes(id));
+    if (idsToAdd.length === 0 && idsToRemove.length === 0) return;
+    const newIndexMap = Object.entries(indexMap)
+      .map(([id, glasbeyIndex]) => (itemIds.includes(id) ? { [id]: glasbeyIndex } : {}))
+      .reduce((a, b) => ({ ...a, ...b }), {});
+
+    let tryIndex = 0;
+    while (idsToAdd.length) {
+      const currentId = idsToAdd.shift();
+      const takenIndices = Object.values(newIndexMap);
+      if (currentId === undefined) continue;
+      while (takenIndices.includes(tryIndex)) tryIndex += 1;
+      newIndexMap[currentId] = tryIndex;
+    }
+
+    setIndexMap(newIndexMap);
+  }, [itemIds, indexMap]);
```

Comment:
because `itemIds` maps `ids` every render, this causes react to render each component that uses this hook twice every time. could we instead use a ref that keeps track of id to glasbey numbers we've seen before that increments based off of the size of the object?

Reply:
the `.map()` was actually redundant, so I removed it. does that resolve the issue?

Reply:
It reduces the double-render issue from every render to every time the ids change, so it should be fine.

Reply:
however, I'm not seeing that removal in the diff -- is it checked in?

Reply:
this is the one im referring to https://github.com/determined-ai/determined/pull/6382/commits/60fee4d827f4a3eecc8a4e6e78f804b9959a654d#diff-e3c320da3a8899401f936d34de98a7ffb335476e6020d54266fef52876eabb7a

in a later commit, i took out the aliasing as well

Reply:
sorry -- must have been a caching issue on my end!